### PR TITLE
Add CBMC proof for vTCPWindowDestroy

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------
This PR adds CBMC proof for vTCPWindowDestroy.

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/TCPWin/vTCPWindowDestroy/vTCPWindowDestroy_harness.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        xTxSegmentListItem[ i ].xSegmentItem.pvOwner = &( xTxSegmentListItem[ i ] );
+
+        /* Assume the container of the queue item is NULL. */
+        __CPROVER_assume( xTxSegmentListItem[ i ].xQueueItem.pxContainer == NULL );
+
+        vListInsertGeneric( &xWindow.xTxSegments, &( xTxSegmentListItem[ i ].xSegmentItem ), &xWindow.xTxSegments.xListEnd );
+    }
+
+    /* Call the function. */
+    vTCPWindowDestroy( &xWindow );
```
Comment (markrtuttle) [inline test/cbmc/proofs/TCPWin/vTCPWindowDestroy/vTCPWindowDestroy_harness.c:59]: Does destroy try to free any of the TPSegment_t objects in the list (eg, does stack or heap allocation matter)?  Does it matter that the lists allocated here are always nonempty?  Do you want the length of the lists to be nondeterministically chosen between 0 and NUM_OF_SEGMENTS?

**Review by markrtuttle — CHANGES_REQUESTED** [review]

File: test/cbmc/proofs/TCPWin/vTCPWindowDestroy/vTCPWindowDestroy_harness.c (hunk 1 above)
Comment (AniruddhaKanhere) [reply test/cbmc/proofs/TCPWin/vTCPWindowDestroy/vTCPWindowDestroy_harness.c:59]: Ah! yes!!
This is something I can do. It does not matter whether the list is always non-empty or empty.

Comment (AniruddhaKanhere) [reply test/cbmc/proofs/TCPWin/vTCPWindowDestroy/vTCPWindowDestroy_harness.c:64]: Done! Thanks!

File: test/cbmc/proofs/TCPWin/vTCPWindowDestroy/vTCPWindowDestroy_harness.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    /* Below loop fills in various segments in the Rx/Tx list of the window. */
+    for( int i = 0; i < NUM_OF_SEGMENTS; i++ )
+    {
+        /********************** Fill in Rx segments ********************/
+        xRxSegmentListItem[ i ].xSegmentItem.pvOwner = &( xRxSegmentListItem[ i ] );
+
+        /* Assume the container of the queue item is NULL. */
+        __CPROVER_assume( xRxSegmentListItem[ i ].xQueueItem.pxContainer == NULL );
+
+        vListInsertGeneric( &xWindow.xRxSegments, &( xRxSegmentListItem[ i ].xSegmentItem ), &xWindow.xRxSegments.xListEnd );
```
[the remark this answers is not in the document]
Comment (AniruddhaKanhere) [reply test/cbmc/proofs/TCPWin/vTCPWindowDestroy/vTCPWindowDestroy_harness.c:51]: I have kept that as is for now.

File: test/cbmc/proofs/TCPWin/vTCPWindowDestroy/vTCPWindowDestroy_harness.c [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    vListInitialise( &xWindow.xTxSegments );
+
+    /* Below loop fills in various segments in the Rx/Tx list of the window. */
+    for( int i = 0; i < NUM_OF_SEGMENTS; i++ )
+    {
+        /********************** Fill in Rx segments ********************/
+        xRxSegmentListItem[ i ].xSegmentItem.pvOwner = &( xRxSegmentListItem[ i ] );
+
+        /* Assume the container of the queue item is NULL. */
+        __CPROVER_assume( xRxSegmentListItem[ i ].xQueueItem.pxContainer == NULL );
```
[the remark this answers is not in the document]
Comment (AniruddhaKanhere) [reply test/cbmc/proofs/TCPWin/vTCPWindowDestroy/vTCPWindowDestroy_harness.c]: Done! Thanks!

File: test/cbmc/proofs/TCPWin/vTCPWindowDestroy/vTCPWindowDestroy_harness.c (hunk 1 above)
Comment (markrtuttle) [reply test/cbmc/proofs/TCPWin/vTCPWindowDestroy/vTCPWindowDestroy_harness.c:64]: Great!  Still want to assume the argument to TCPWindowDestroy is not NULL?

**Review by markrtuttle — DISMISSED** [review]
Left one comment, but otherwise great!

**Review by n9wxu — APPROVED** [review]

**Review by markrtuttle — APPROVED** [review]

**Pull request merged**